pkg/destroy/aws/aws.go

```diff
@@ -140,16 +140,28 @@ func (o *ClusterUninstaller) Run() error {
 						&resourcegroupstaggingapi.GetResourcesInput{TagFilters: tagFilters},
 						func(results *resourcegroupstaggingapi.GetResourcesOutput, lastPage bool) bool {
 							for _, resource := range results.ResourceTagMappingList {
-								arn := *resource.ResourceARN
-								if _, ok := deleted[arn]; !ok {
+								arnString := *resource.ResourceARN
+								if _, ok := deleted[arnString]; !ok {
+									arnLogger := o.Logger.WithField("arn", arnString)
 									matched = true
-									err := deleteARN(awsSession, arn, filter, o.Logger)
+									parsed, err := arn.Parse(arnString)
 									if err != nil {
-										err = errors.Wrapf(err, "deleting %s", arn)
-										o.Logger.Debug(err)
+										arnLogger.Debug(err)
 										continue
 									}
-									deleted[arn] = exists
+
+									if parsed.Service == "ec2" && strings.HasPrefix(parsed.Resource, "network-interface/") {
```

Comment:
why is this not handled in the deleteARN function? we could easily skip here https://github.com/wking/openshift-installer/blob/fc70eb221cd211402c6f0c13e679cbee9d071ae4/pkg/destroy/aws/aws.go#L522
?

i don' think we should keep these conditionals in main loop..

Reply:
> why is this not handled in the deleteARN function?

Because I want `deleteARN` to actually delete what you asked it to ;).  Here we know we got the ARN from a direct tag.  Inside `deleteARN` we don't.

Reply:
move this into deleteTaggedResource like func which does conditional before calling delete ARN.

_the main loop is not correct place to keep this conditional_